src/resolve/resolved-manager.c

```diff
@@ -1054,6 +1054,58 @@ static int manager_ipv6_send(
         return sendmsg_loop(fd, &mh, 0);
 }
 
+int send_dns_notification(Manager *m, DnsAnswer *answer, const char *query_name) {
+        _cleanup_free_ char *normalized = NULL;
+        DnsResourceRecord *rr;
+        int ifindex, r;
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        Varlink *connection;
+
+        assert(m);
+
```

Comment:
Please return earlier if there is no subscriber.

Reply:
Added a check